skyrl-tx/tx/tinker/engine.py

```diff
@@ -617,6 +617,9 @@ def process_sample_batch(
         if not valid_requests:
             return results
 
+        # Computes prompt_logprobs for the whole batch if any request asked for them
+        needs_prompt_logprobs = any(request_data.prompt_logprobs for (_, request_data) in valid_requests.values())
```

Comment:
Probably this should be done per micro-batch going forward to avoid unnecessary computations, and I was also a little confused why it was
```
                if needs_prompt_logprobs and result.prompt_logprobs:
                    all_prompt_logprobs.extend(result.prompt_logprobs[:batch_size])
```
below and not (unconditionally)
```
all_prompt_logprobs.extend(result.prompt_logprobs[:batch_size] if result.prompt_logprobs else [None] * batch_size)
```
but it makes sense since `needs_prompt_logprobs` applies for the whole batch :)

Reply:
For now totally fine to keep it simple as it is :)